keras_nlp/samplers/top_k_sampler_test.py

```diff
@@ -16,156 +16,95 @@
 import numpy as np
 import tensorflow as tf
 from absl.testing import parameterized
-from tensorflow import keras
 
 from keras_nlp.samplers.top_k_sampler import TopKSampler
 
 
 class TopKSamplerTest(tf.test.TestCase, parameterized.TestCase):
     def setUp(self):
         super().setUp()
-        self.vocab_size = 10
-        self.feature_size = 16
-
-        # Create a dummy model to predict the next token.
-        model = keras.Sequential(
-            [
-                keras.Input(shape=[None]),
-                keras.layers.Embedding(
-                    input_dim=self.vocab_size,
-                    output_dim=self.feature_size,
-                ),
-                keras.layers.Dense(self.vocab_size),
-                keras.layers.Softmax(),
-            ]
+        # Use a simple alphabet of lowercase characters to [0, 26).
+        self.int_lookup = {i: chr(i + ord("a")) for i in range(26)}
+        self.char_lookup = {v: k for k, v in self.int_lookup.items()}
+        self.batch_size = 1
+        self.length = 12
+        self.vocab_size = len(self.int_lookup)
+
+        def next(prompt, state, index):
+            # Return a probability distribution favoring the next char in state.
+            logits = tf.one_hot(state[:, index], self.vocab_size) * 1e9
+            return logits, state
+
+        self.next = next
+        self.sampler = TopKSampler(k=5)
+
+    def join_as_string(self, x):
+        return ["".join([self.int_lookup[i] for i in s]) for s in x.numpy()]
+
+    def test_stateless(self):
```

Comment:
Optional nit: `test_stateful_call`/`test_stateful_sample`? test + an adjective looks a bit strange to me.